### PR TITLE
Cdg turbosub - Add new romsets for TurboSub and add start inputs back to new prototype

### PR DESCRIPTION
**Pull request by lupinthird**

### CODE REVIEW

**Comment by lupinthird** [comment]
These are the matching TurboSub driver updates for the roms we're sending over to the submission account

**Comment by simzy39** [comment]
Are these roms from the collector that owns the turbosub dot com website?

**Comment by lupinthird** [comment]
Yes, Chris runs the website and he sent them to his cousin (Spindaddy).  These are being released with their knowledge and consent. I worked with Spindaddy to get them working in mame and to update the driver.

**Review by philipjbennett — CHANGES_REQUESTED** [review]

File: src/mame/misc/esripsys.cpp [hunk 1]
```diff
@@ -459,6 +459,48 @@ static INPUT_PORTS_START( turbosub )
 	PORT_BIT( 0xff, 0x80, IPT_AD_STICK_Y ) PORT_SENSITIVITY(70) PORT_KEYDELTA(30)
 INPUT_PORTS_END
 
+static INPUT_PORTS_START( turbosubb8 )
```
[the remark this answers is not in the document]
Comment (lupinthird) [reply src/mame/misc/esripsys.cpp:462]: This should look much more cleaned up now

File: src/mame/misc/esripsys.cpp [hunk 2]
```diff
@@ -1062,6 +1104,463 @@ ROM_START( turbosub6 )
...
+	ROM_LOAD( "6331_vid.u155", 0x0240, 0x0020, CRC(63371737) SHA1(f08c03c81322c0de9ee64b4a9f11a1422c5bd463) )
+ROM_END
+
+ROM_START( turbosubb8 )
+	ROM_REGION( 0xc0000, "main_code", 0) /* Non-bankswitched, 6809 #0 code */
+	ROM_LOAD( "tsb8.u85",    0x18000, 0x4000, CRC(cd920709) SHA1(50c7755be4c37b1b59660c28b859293a71017bc3) )
+
+	ROM_REGION( 0x48000, "game_cpu", 0 ) /* Bankswitched 6809 code */
+	ROM_LOAD( "tsb8.u82",    0x10000, 0x2000, CRC(9af7c67c) SHA1(55e243c05989cbba5a60404f699292adf4e389e6) )
+	ROM_CONTINUE(                0x20000, 0x2000 )
```
[the remark this answers is not in the document]
Comment (lupinthird) [reply src/mame/misc/esripsys.cpp]: These are now aligned as they are everywhere else

File: src/mame/misc/esripsys.cpp [hunk 3]
```diff
@@ -1070,5 +1569,9 @@ ROM_END
  *************************************/
 
 GAME( 1985, turbosub,  0,        esripsys, turbosub, esripsys_state, init_esripsys, ROT0, "Entertainment Sciences", "Turbo Sub (prototype rev. TSCA)", MACHINE_SUPPORTS_SAVE )
```
[the remark this answers is not in the document]
Comment (lupinthird) [reply src/mame/misc/esripsys.cpp]: These are all neatly aligned again

**Review by philipjbennett — APPROVED** [review]

**Comment by simzy39** [comment]
Can this comment now be removed from the file?
Roms Wanted:  Turbo Sub [later version] (improved gameplay, uses 27512 ROMs)"

**Comment by lupinthird** [comment]
The turbosub8 set which is the newest release had graphics ROMs that were on 27256s, but the program code was still on 27128s. There may still be a different version of larger combined ROMs in the wild.

**Comment by lupinthird** [comment]
Most likely turbosub8 was the final release, so maybe? How do we know of the existence of the 27512 ROMs?

**Pull request merged**